rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -1750,10 +1702,9 @@ extern "C"
             {
                 void *data = guard_conditions->guard_conditions[i];
                 GuardCondition *guard_condition = (GuardCondition*)data;
-                if(!guard_condition->getHasTriggered())
-                {
-                    guard_conditions->guard_conditions[i] = 0;
-                }
+                // Guard conditions persist (don't set to null)
+                // We call getHasTriggered to reset the trigger state.
```

Comment:
Without this we are not able to check that our code is sleeping correctly. Simply because of that I think it is crucial to keep.

Second: the only reason why our executor has to refill all the handles every time before wait is because it simply doesn't use the information if the interrupt condition was triggered (in order to use new/changes handles). I think that is a good example why using this information is useful to avoid overhead like this.